sinks/signalfx/signalfx_test.go

```diff
@@ -46,6 +47,13 @@ func TestNewSignalFxSink(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	httpsink, ok := sink.client.(*sfxclient.HTTPSink)
+	if !ok {
+		assert.Fail(t, "SignalFX sink isn't the correct type")
```

Comment:
Unless this is the only possible error type, this seems like it could be a really misleading failure message -- why not log the error message itself? (Possibly with some extra context)

Reply:
That's the only error type, afaik. This is effectively "Is this a an `sfxclient.HTTPSink`?" wherein `httpsink` is now defined as such and `ok` is true or false depending on the success of the "cast".

Reply:
That's correct (although with a nit: that's a type assertion, not a cast). `ok` is a bool, so there's no other message to log.